Dummy.h

```diff
@@ -0,0 +1,97 @@
+#ifndef SAILGAME_TEXAS
+#define SAILGAME_TEXAS
+
+#include <iosfwd>
+#include <map>
+#include <string>
+#include <vector>
+
+struct GameStatus;
+struct CardScore;
+
+class Dummy {
+public:
+  using uid_t = int;
+  using chip_t = int;
+  using card_t = int;
+  using status_t = int;
+  using score_t = CardScore;
+
+  enum { STOP = -1, READY = 0, NOT_YOUR_TURN = 1, INVALID_BET = 2 };
+
+private:
+  std::map<uid_t, const std::string> uid2addr;
+
+  std::vector<chip_t> chips;
+
+  std::map<uid_t, std::vector<card_t>> holecards;
+  std::map<uid_t, chip_t> bankroll, roundbets;
+  std::vector<card_t> board, deck;
+
+  status_t state;
+  int user_count;
+  uid_t prev_pos, button, prev_winner, cutoff;
+  chip_t cur_chips;
+  bool raised;
+
+public:
+  explicit Dummy()
+      : state(STOP), user_count(0), prev_pos(-1), button(-1), prev_winner(-1),
+        cutoff(-1), cur_chips(-1), raised(false) {}
+
+  const status_t Begin();
+  const uid_t Join(const std::string &addr);
+  // positive -> raise or call; zero -> check; -1 -> conceed.
+  const status_t Play(const uid_t uid, const chip_t value);
+
+  const GameStatus DumpStatusForUser(const uid_t uid) const;
+  const void DumpDebugMessages(std::ostream &out);
+
+  const uid_t PreviousWinner() const { return prev_winner; }
+  const uid_t FirstPlayer() const { return 1; }
```

Comment:
should the firstplayer and lastplayer consider someone all-in or fold?
say if the uid == 1 is all-in,  we should skip the execution? do you handle this case?

Reply:
that's still in todo-list.
working on score logic currently.